tensorboard/webapp/feature_flag/effects/feature_flag_effects.ts

```diff
@@ -0,0 +1,52 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+import {Injectable} from '@angular/core';
+import {createAction, Action} from '@ngrx/store';
+import {Actions, ofType, createEffect} from '@ngrx/effects';
+import {map} from 'rxjs/operators';
+
+import {TBFeatureFlagDataSource} from '../../webapp_data_source/tb_feature_flag_data_source_types';
+import {featuresLoaded} from '../actions/feature_flag_actions';
+
+/** @typehack */ import * as _typeHackRxjs from 'rxjs';
+/** @typehack */ import * as _typeHackNgrx from '@ngrx/store/src/models';
+/** @typehack */ import * as _typeHackNgrxEffects from '@ngrx/effects';
+
+const effectsInitialized = createAction('[FEATURE FLAG] Effects Init');
```

Comment:
This action looks a little peculiar. Who / what is supposed to fire this action (outside tests)?

Reply:
Ngrx, on effects initialization, calls `ngrxOnInitEffects` gets this action's payload. It, then, puts the payload into the action stream so we can write effects that do stuff on the initialization.